app/controllers/followup_controller.rb

```diff
@@ -0,0 +1,189 @@
+      return
+    end
+  end
+
+  def check_for_external_request
+    if @info_request.is_external?
+      @reason = 'external'
+      render :action => 'followup_bad'
+      return
+    end
+  end
+
+  def check_internal_review
+    @internal_review = false
+    @internal_review_pass_on = false
+    if params[:internal_review]
+      @internal_review = true
+      @internal_review_pass_on = true
+    end
+  end
+
+  def check_reedit
+    if params[:reedit]
+      render :action => 'new'
+      return
+    end
+  end
+
+  def check_request_matches_incoming_message
+    if @incoming_message and @info_request != @incoming_message.info_request
+      raise ActiveRecord::RecordNotFound.
+              new("Incoming message #{@incoming_message.id} does not belong " \
+                    "to request #{@info_request.id}")
+    end
+  end
+
+  def check_responses_allowed
+    if @info_request.allow_new_responses_from == "nobody"
+      flash[:error] = _('Your follow up has not been sent because this request has been stopped to prevent spam. Please <a href="{{url}}">contact us</a> if you really want to send a follow up message.', :url => help_contact_path.html_safe)
+      render :action => 'new'
+      return
+    end
+  end
+
+  def check_user_credentials
+    # We want to make sure they're the right user first, before they start
+    # writing a message and wasting their time if they are not the requester.
+    params = get_login_params(@incoming_message)
+    return if !authenticated_as_user?(@info_request.user, params)
+    if authenticated_user and !authenticated_user.can_make_followup?
+      @details = authenticated_user.can_fail_html
+      render :template => 'user/banned'
+      return
+    end
+    if authenticated_user && !@info_request.user_can_view?(authenticated_user)
+      return render_hidden
+    end
+  end
+
+  def get_login_params(is_incoming)
+    if is_incoming
+      { :web => _("To send a follow up message to {{authority_name}}",
+                  :authority_name => @info_request.public_body.name),
+        :email => _("Then you can write follow up message to {{authority_name}}.",
+                    :authority_name => @info_request.public_body.name),
+        :email_subject => _("Write your FOI follow up message to {{authority_name}}",
+                            :authority_name => @info_request.public_body.name) }
+    else
+      { :web => _("To reply to {{authority_name}}.",
+                  :authority_name => @info_request.public_body.name),
+        :email => _("Then you can write your reply to {{authority_name}}.",
+                    :authority_name => @info_request.public_body.name),
+        :email_subject => _("Write a reply to {{authority_name}}",
+                            :authority_name => @info_request.public_body.name) }
+    end
+  end
+
+  def outgoing_message_params
+    params_outgoing_message = params[:outgoing_message] ? params[:outgoing_message].clone : {}
+    params_outgoing_message.merge!({
```

Comment:
Use `params_outgoing_message[:status] = 'ready'; params_outgoing_message[:message_type] = 'followup'; params_outgoing_message[:incoming_message_followup] = @incoming_message; params_outgoing_message[:info_request_id] = @info_request.id` instead of `params_outgoing_message.merge!({
                                     :status => 'ready',
                                     :message_type => 'followup',
                                     :incoming_message_followup => @incoming_message,
                                     :info_request_id => @info_request.id
    })`.